plugins/logitech-bulkcontroller/fu-plugin-logitech-bulkcontroller.c

```diff
@@ -0,0 +1,388 @@
+	g_cond_init(&plugin_data->test_upd_cond);
+	g_mutex_init(&plugin_data->test_upd_mutex);
+	g_mutex_lock(&plugin_data->test_upd_mutex);
+
+	ret = proto_manager_generate_get_device_info_request(&message);
+	if (!ret && message.data) {
+		ReturnValue *ret_val;
+		g_debug("Sending GetDeviceInfoRequest. Length %lu", message.len);
+		ret_val = logibulkcontroller_send_data_sync(plugin_data->bulkcontroller_obj,
+							    message.data,
+							    message.len);
+		g_free(message.data);
+		if ((ERRORCODE_NO_ERROR != ret_val->error_code) &&
+		    (ERRORCODE_SEND_DATA_REQUEST_PUSHED_TO_QUEUE != ret_val->error_code)) {
+			g_warning("Error in sending GetDeviceInfoRequest %u", ret_val->error_code);
+			return FALSE;
+		}
+		if (g_cond_wait_until(&plugin_data->test_upd_cond,
+				      &plugin_data->test_upd_mutex,
+				      g_get_monotonic_time() + 5 * G_TIME_SPAN_SECOND)) {
+			g_mutex_unlock(&plugin_data->test_upd_mutex);
+			g_mutex_lock(&plugin_data->test_upd_mutex);
+		}
+	}
+	return TRUE;
+}
+
+void
+fu_plugin_init(FuPlugin *plugin)
+{
+	g_debug("Initializing Logitech bulk controller plugin");
+	fu_plugin_set_build_hash(plugin, FU_BUILD_HASH);
+	fu_plugin_alloc_data(plugin, sizeof(FuPluginData));
+}
+
+gboolean
+fu_plugin_coldplug(FuPlugin *plugin, GError **error)
+{
+	LogiBulkController *obj;
+	FuPluginData *plugin_data;
+	gint pid;
+	FuContext *ctx;
+	BulkControllerCallbacks bulkcb = {bulk_error_cb,
+					  bulk_file_transfer_cb,
+					  read_upd_data_cb,
+					  read_sync_data_cb,
+					  send_data_sync_cb};
+
+	plugin_data = fu_plugin_get_data(plugin);
+	plugin_data->bulkcontroller_obj = NULL;
+	/* Check if Logitech Rally Bar Mini is conneced */
+	pid = 0x08D3;
+	plugin_data->device_name = g_strdup("Rally Bar Mini");
+	plugin_data->device_type = kDeviceTypeRallyBarMini;
+	plugin_data->device_info_available = FALSE;
+	plugin_data->run_devicemode_info_ = FALSE;
+	obj = logibulkcontroller_create_bulk_controller(0x46d, pid, bulkcb, plugin_data);
+	if (ERRORCODE_NO_ERROR == logibulkcontroller_open_device(obj)) {
+		g_autoptr(FuDevice) device = NULL;
+		plugin_data->bulkcontroller_obj = obj;
+		if (!send_device_info_request(plugin_data)) {
+			g_warning(" Failed to communicate with device %s",
+				  plugin_data->device_name);
+			logibulkcontroller_close_device(obj);
+			return FALSE;
+		}
+		/* Wait for the response */
+		sleep(TIMEOUT_SYNC_RESPONSE_5SEC);
+		if (!plugin_data->device_info_available) {
+			g_warning(" Failed to receive device information from device %s",
+				  plugin_data->device_name);
+			logibulkcontroller_close_device(obj);
+			return FALSE;
+		}
+		ctx = fu_plugin_get_context(plugin);
+
+		device = fu_device_new_with_context(ctx);
+		fu_device_set_id(device, "logibulkctrl-video-conf-devices");
+		fu_device_add_guid(device, "7da607b9-1f8f-597a-8f5e-5788d6f8bcb3");
+		fu_device_set_name(device, plugin_data->device_info.name);
+		fu_device_set_version_format(device, FWUPD_VERSION_FORMAT_TRIPLET);
```

Comment:
since you have this in a quirk file, I don't think you need it here too

Reply:
Removed in the quirk file